src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentSelector.cs

```diff
@@ -21,49 +23,30 @@ public DefaultViewComponentSelector(IAssemblyProvider assemblyProvider)
 
         public Type SelectComponent([NotNull] string componentName)
         {
-            var assemblies = _assemblyProvider.CandidateAssemblies;
-            var types = assemblies.SelectMany(a => a.DefinedTypes);
+            if (_cache == null)
+            {
+                var assemblies = _assemblyProvider.CandidateAssemblies;
+                var types = assemblies.SelectMany(a => a.DefinedTypes);
```

Comment:
No, this is going to mess with stuff because of Type/TypeInfo. I don't want to mess with that in a perf change.